Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp

```diff
@@ -0,0 +1,324 @@
+        physx::PxArticulationLink* parentLink, const ArticulationLinkData& thisLinkData)
+    {
+        using namespace physx;
+
+        const Physics::ColliderConfiguration& colliderConfiguration = thisLinkData.m_colliderConfiguration;
+        const Physics::ShapeConfiguration* shapeConfiguration = thisLinkData.m_shapeConfiguration.get();
+
+        AZStd::shared_ptr<Physics::Shape> physicsShape;
+
+        if (shapeConfiguration)
+        {
+            if (shapeConfiguration->GetShapeType() == Physics::ShapeType::PhysicsAsset)
+            {
+                const auto* physicsAssetShapeConfiguration =
+                    static_cast<const Physics::PhysicsAssetShapeConfiguration*>(shapeConfiguration);
+                if (!physicsAssetShapeConfiguration->m_asset.IsReady())
+                {
+                    const_cast<Physics::PhysicsAssetShapeConfiguration*>(physicsAssetShapeConfiguration)->m_asset.BlockUntilLoadComplete();
+                }
+
+                const bool hasNonUniformScale = (AZ::NonUniformScaleRequestBus::FindFirstHandler(GetEntityId()) != nullptr);
+                AZStd::vector<AZStd::shared_ptr<Physics::Shape>> assetShapes;
+                Utils::CreateShapesFromAsset(
+                    *physicsAssetShapeConfiguration,
+                    colliderConfiguration,
+                    hasNonUniformScale,
+                    physicsAssetShapeConfiguration->m_subdivisionLevel,
+                    assetShapes);
+
+                physicsShape = assetShapes[0];
+            }
+            else
+            {
+                Physics::SystemRequestBus::BroadcastResult(
+                    physicsShape, &Physics::SystemRequests::CreateShape, colliderConfiguration, *shapeConfiguration);
+            }
+
+            m_articulationShapes.emplace_back(physicsShape);
+        }
+
+        PxTransform thisLinkTransform;
+        if (parentLink)
+        {
+            PxTransform parentLinkTransform = parentLink->getGlobalPose();
+            PxTransform thisLinkRelativeTransform = PxMathConvert(thisLinkData.m_relativeTransform);
+            thisLinkTransform = parentLinkTransform * thisLinkRelativeTransform;
+        }
+        else
+        {
+            thisLinkTransform = PxMathConvert(GetEntity()->GetTransform()->GetWorldTM());
+        }
+
+        PxArticulationLink* thisLink = m_articulation->createLink(parentLink, thisLinkTransform);
+        if (!thisLink)
+        {
+            AZ_Error("PhysX", false, "Failed to create articulation link at root %s", GetEntity()->GetName().c_str());
+            return;
+        }
+
+        // Setup actor data
+        AZStd::shared_ptr<ActorData> thisLinkActorData = AZStd::make_shared<ActorData>(thisLink);
+        thisLinkActorData->SetEntityId(thisLinkData.m_entityId);
+        m_linksActorData.emplace_back(thisLinkActorData);
+
+        if (parentLink)
+        {
+            PxArticulationJointReducedCoordinate* inboundJoint = thisLink->getInboundJoint();
```

Comment:
I had problem compiling here, CLang was unhappy with implicit casting.
I would use PhysX casting here : 
```suggestion
            PxArticulationJointReducedCoordinate* inboundJoint = thisLink->getInboundJoint()->is<PxArticulationJointReducedCoordinate>();
            AZ_Assert(inboundJoint, "It is not PxArticulationJointReducedCoordinate");
```